tests/testthat/test-loopy-logging.R

```diff
@@ -96,8 +101,136 @@ test_that("prediction error doesn't stop grid", {
   )
 
   expect_identical(
-    res_fit, 
+    dplyr::select(res_fit, -.notes), 
     exp,
     ignore_attr = TRUE
   )
+
+  expect_identical(nrow(res_fit$.notes[[1]]), 2L)
+  expect_identical(ncol(res_fit$.notes[[1]]), 3L)
+  expect_true(all(vapply(res_fit$.notes[[1]], is.character, logical(1))))
+})
+
+test_that("capturing error correctly in notes", {
+  assign("prep.step_logging_helper", prep.step_logging_helper, envir = .GlobalEnv)
+  assign("bake.step_logging_helper", bake.step_logging_helper, envir = .GlobalEnv)
```

Comment:
i don't like this, but for now it is an easy way to test conditions of different types